### PR TITLE
[SPARK-44525][SQL] Improve error message when Invoke method is not found

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

This PR aims to improve the error message when `Invoke`'s `method` is not found.

### Why are the changes needed?

Currently, the error message is not clear when `Invoke`'s `method` is not found.

There is one error message I have encountered, which is not much helpful to find the root cause.
```
org.apache.spark.SparkException: [INTERNAL_ERROR] A method named "invoke" is not declared in any enclosing class nor any supertype
	at org.apache.spark.SparkException$.internalError(SparkException.scala:77)
	at org.apache.spark.SparkException$.internalError(SparkException.scala:81)
	at org.apache.spark.sql.errors.QueryExecutionErrors$.methodNotDeclaredError(QueryExecutionErrors.scala:452)
	at org.apache.spark.sql.catalyst.expressions.objects.InvokeLike.findMethod(objects.scala:173)
	at org.apache.spark.sql.catalyst.expressions.objects.InvokeLike.findMethod$(objects.scala:170)
	at org.apache.spark.sql.catalyst.expressions.objects.Invoke.findMethod(objects.scala:363)
	at org.apache.spark.sql.catalyst.expressions.objects.Invoke.method$lzycompute(objects.scala:391)
	at org.apache.spark.sql.catalyst.expressions.objects.Invoke.method(objects.scala:389)
	at org.apache.spark.sql.catalyst.expressions.objects.Invoke.eval(objects.scala:401)
	at org.apache.spark.sql.catalyst.expressions.HashExpression.eval(hash.scala:292)
	at org.apache.spark.sql.catalyst.expressions.Pmod.eval(arithmetic.scala:1054)
```

### Does this PR introduce _any_ user-facing change?

Yes, Spark returns a more clear error message.

### How was this patch tested?

Add UT.

### CODE REVIEW

**Comment by pan3793** [comment]
cc @HyukjinKwon @yaooqinn @LuciferYang

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 1]
```diff
@@ -465,6 +465,20 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
       s"""A method named "$name" is not declared in any enclosing class nor any supertype""")
   }
 
+  def methodNotFoundError(
```
Comment (LuciferYang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:468]: this was defined as `internalError` previously, really need to add a new `errorClass`?

Comment (pan3793) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:468]: I searched the codebase, seems `Invoke` `StaticInvoke` is only used internally, let me convert it to `internalError` as you suggested.

Comment (pan3793) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:468]: updated

**Review by LuciferYang — APPROVED** [review]

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala [hunk 2]
```diff
@@ -73,6 +73,32 @@ class ObjectExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
...
+    val e2 = intercept[SparkException] {
+      Invoke(
+        inputObject,
+        "oneArgNotExistMethod",
+        IntegerType,
+        Seq(Literal.fromObject(UTF8String.fromString("dummyInputString"))),
+        Seq(StringType)).eval(inputRow)
+    }
+
+    assert(e2.getErrorClass == "INTERNAL_ERROR")
```
[the remark this answers is not in the document]
Comment (pan3793) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala:97]: thought, but it seems does not support verifying the message

[the remark this answers is not in the document]
Comment (pan3793) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala:97]: Thanks, updated

**Review by LuciferYang — APPROVED** [review]

**Comment by pan3793** [comment]
Emm... seems the test passed but reported failure? https://github.com/pan3793/spark/actions/runs/5653530123

**Review by ulysses-you — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged into master 3.5. Thanks @ulysses-you and @pan3793

**Comment by pan3793** [comment]
Thanks @LuciferYang and @ulysses-you

**Pull request closed without merging**